addons/hr_holidays/models/hr_leave.py

```diff
@@ -25,6 +25,13 @@ class HolidaysRequest(models.Model):
     def _default_employee(self):
         return self.env.context.get('default_employee_id') or self.env['hr.employee'].search([('user_id', '=', self.env.uid)], limit=1)
 
+    def _default_holiday_status_id(self):
+        LeaveType = self.env['hr.leave.type'].with_context(employee_id=self._default_employee().id)
+        lt = LeaveType.search([('valid', '=', True), ('employee_applicability', 'in', ['leave', 'both']), ('limit', '=', False)])
```

Comment:
To test: one search with order="limit ASC (or DESC ?)" to have one search to find the correct leave type. As we need only on e result (the first one) you can add the limit parameter. Side note: check what happend using order if we have NULL, False and True values. Good luck. If null results are always first it should be ok to have null, then false, then true in the ordered results.